rs/registry/transport/src/lib.rs

```diff
@@ -113,6 +113,38 @@ impl From<Error> for RegistryError {
     }
 }
 
+impl registry_mutation::Type {
+    pub fn is_delete(self) -> bool {
+        // Do not simply replace this with self == Delete, because that assumes
+        // the reader knows that all other mutation types are not some other
+        // flavor of deletion. Whereas, this match proves (in writing) that we
+        // really individually considered every single mutation types.
+        match self {
+            registry_mutation::Type::Delete => true,
+
+            registry_mutation::Type::Insert
+            | registry_mutation::Type::Update
+            | registry_mutation::Type::Upsert => false,
+        }
+    }
+
+    /// Returns whether key being touched
+    ///
+    ///   * must be present. In this case, Some(true) is returned.
+    ///   * must be absence. In this case, Some(false) is returned.
+    ///   * can be present or absent: In this case None is returned.
+    pub fn requires_already_present(self) -> Option<bool> {
```

Comment:
```suggestion
    pub fn requires_key_already_present(self) -> Option<bool> {
```

Reply:
Moot: I changed the name of this to presence_requirement, because I made it return a PresenceRequirement (enum, per other feedback).